File: 1ºAno/Lista3/C/Casa/files/ex2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    
    for (;;) {
        char *dados = (char*)malloc(10 * sizeof(char));
        int vals[3] = {0, 0, 0};
        for (;;) {
            printf ("Na mesma linha, entre, respectivamente, com os valores inteiros referentes a:\n\n1) Primeiro Número (a1)\n2) Último número (an | an < a1)\n3) Paridade dos termos (2 se forem pares e 1 se forem ímpares)\n\n");
            fgets (dados, 10, stdin);
            char *pt = strtok(dados, " ");
            int i = 0;
            while (pt != NULL) {
                vals[i] = atoi(pt);
                ++i;
                pt = strtok(NULL, " ");
            }
            if (vals[2] == 0 && vals[1] != vals[0]) {
                printf ("*Número(s) inválidos... Tente novamente\n*");
            }
            else {
                break;
            }
        }
        if (vals[0] == 0 && vals[1] == 0 && vals[2] == 0) {
            break;
        }
        if (vals[0] % 2 == 0 && vals[2] == 1 || vals[0] % 2 != 0 && vals[2] == 2) {
            vals[0] -= 1;
        }
        if (vals[1] % 2 == 0 && vals[2] == 1 || vals[1] % 2 != 0 && vals[2] == 2) {
            vals[1] -= 1;
        }
        printf ("Aqui estão os termos x | x %% 2 = %d, entre %d e %d:\n", vals[2], vals[0], vals[1]);
        while (vals[0] >= vals[1]) {
            printf ("%d ", vals[0]);
            vals[0] -= 2;
        }
        printf ("\n");
    }
    
    return 0;
}

File: 1ºAno/Lista3/Python/Casa/files/ex2.py
while True:
    while True:
        try:
            vals = input("Na mesma linha, entre com os valores, respectivamente, referentes a:\n\n1) 1º termo da sequência (a1)\n2) Último termo da sequência (an)\n3) Paridade dos termos (0 para pares e 1 para ímpares)\nPara sair, entre com 0 em todos os valores\n\n").split()
            vals[0], vals[1], vals[2] = int(vals[0]), int(vals[1]), int(vals[2])
            if vals[2] != 1 and vals[2] != 0:
                print ("*Número(s) inválido(s)... Tente novamente*")
            else:
                break
        except:
            print ("*Número(s) inválido(s)... Tente novamente*")
    if vals[0] == vals[1] == vals[2] == 0:
        break
    if vals[2] == 0:
        print (f"Aqui estão os termos pares, entre {vals[0]} e {vals[1]}:")
    else:
        print (f"Aqui estão os termos ímpares, entre {vals[0]} e {vals[1]}:")
    if vals[0] <= vals[1]:
        if vals[0] % 2 != 0 and vals[2] == 0 or vals[0] % 2 == 0 and vals[2] == 1:
            vals[0] += 1
        if vals[1] % 2 != 0 and vals[2] == 0 or vals[1] % 2 == 0 and vals[2] == 1:
            vals[1] -= 1
        while vals[0] <= vals[1]:
            print (f"{vals[0]} ", end="")
            vals[0] += 2
    else:
        if vals[0] % 2 != 0 and vals[2] == 0 or vals[0] % 2 == 0 and vals[2] == 1:
            vals[0] -= 1
        if vals[1] % 2 != 0 and vals[2] == 0 or vals[1] % 2 == 0 and vals[2] == 1:
            vals[1] += 1
        while vals[0] >= vals[1]:
            print (f"{vals[0]} ", end="")
            vals[0] -= 2
    print ("\n")
print ("\nAdeus!")